runtime/queries/metricsview_timeseries.go

```diff
@@ -340,16 +342,62 @@ func (q *MetricsViewTimeSeries) buildDuckDBSQL(args []any, mv *runtimev1.Metrics
 		shift = fmt.Sprintf("%d MONTH", offset)
 	}
 
-	sql := fmt.Sprintf(
-		`SELECT timezone(?, date_trunc('%[1]s', timezone(?, %[2]s::TIMESTAMPTZ) + INTERVAL %[7]s) - INTERVAL %[7]s) as %[3]s, %[4]s FROM %[5]s WHERE %[6]s GROUP BY 1 ORDER BY 1`,
-		dateTruncSpecifier,             // 1
-		safeName(mv.TimeDimension),     // 2
-		tsAlias,                        // 3
-		strings.Join(selectCols, ", "), // 4
-		safeName(mv.Table),             // 5
-		whereClause,                    // 6
-		shift,                          // 7
-	)
+	sql := ""
+	if shift == "" {
+		if q.TimeGranularity == runtimev1.TimeGrain_TIME_GRAIN_HOUR {
```

Comment:
I think the use of `time_bucket` might be needed for all grains of hour or less (i.e. also minutes, seconds). Otherwise, it will emit just one spiked bin instead of two at DST:

See this example:

```
D select timezone('Europe/Copenhagen', date_trunc('MINUTE', timezone('Europe/Copenhagen', '2023-10-29T00:01:10Z'::TIMESTAMPTZ))) as t;
┌──────────────────────────┐
│            t             │
│ timestamp with time zone │
├──────────────────────────┤
│ 2023-10-29 01:01:00+00   │
└──────────────────────────┘
D select timezone('Europe/Copenhagen', date_trunc('MINUTE', timezone('Europe/Copenhagen', '2023-10-29T01:01:10Z'::TIMESTAMPTZ))) as t;
┌──────────────────────────┐
│            t             │
│ timestamp with time zone │
├──────────────────────────┤
│ 2023-10-29 01:01:00+00   │
└──────────────────────────┘
```

(notice the input UTC timestamps are one hour apart, but the output timestamp is the same)